l5kit/l5kit/environment/envs/l5_env.py

```diff
@@ -0,0 +1,323 @@
+        self.observation_space = spaces.Dict({'image': spaces.Box(low=0, high=1, shape=obs_shape, dtype=np.float32)})
+
+        # Simulator Config within Gym
+        self.sim_cfg = sim_cfg if sim_cfg is not None else SimulationConfigGym()
+        self.simulator = ClosedLoopSimulator(self.sim_cfg, self.dataset, device=torch.device("cpu"),
+                                             mode=ClosedLoopSimulatorModes.GYM)
+
+        self.reward = reward if reward is not None else CLEReward()
+
+        self.max_scene_id = cfg["gym_params"]["max_scene_id"]
+        self.overfit = cfg["gym_params"]["overfit"]
+        if self.overfit:
+            self.overfit_scene_id = cfg["gym_params"]["overfit_id"]
+
+        self.cle = cle
+        self.rescale_action = rescale_action
+        self.use_kinematic = use_kinematic
+
+        self.action_rescale_params = self._get_action_rescale_params()
+
+        if self.use_kinematic:
+            self.kin_model = kin_model if kin_model is not None else UnicycleModel()
+
+        # If not None, reset_scene_id is the scene_id that will be rolled out when reset is called
+        self.reset_scene_id = reset_scene_id
+        if self.overfit:
+            self.reset_scene_id = self.overfit_scene_id
+
+        # flag to decide whether to return any info at end of episode
+        # helps to limit the IPC
+        self.return_info = return_info
+
+        self.seed()
+
+    def seed(self, seed: int = None) -> List[int]:
+        """ Generate the random seed.
+
+        :param seed: the seed integer
+        :return: the output random seed
+        """
+        self.np_random, seed = seeding.np_random(seed)
+        # TODO : add a torch seed for future
+        return [seed]
+
+    def reset(self) -> Dict[str, np.ndarray]:
+        """ Resets the environment and outputs first frame of a new scene sample.
+
+        :return: the observation of first frame of sampled scene index
+        """
+        # Define in / outs for new episode scene
+        self.agents_ins_outs: DefaultDict[int, List[List[UnrollInputOutput]]] = defaultdict(list)
+        self.ego_ins_outs: DefaultDict[int, List[UnrollInputOutput]] = defaultdict(list)
+
+        if self.reset_scene_id is not None:
+            self.scene_index = self.reset_scene_id
+        else:  # Sample a scene
+            self.scene_index = random.randint(0, self.max_scene_id)
+        self.sim_dataset = SimulationDataset.from_dataset_indices(self.dataset, [self.scene_index], self.sim_cfg)
+
+        # Reset CLE evaluator
+        self.reward.reset()
+
+        # Output first observation
+        self.frame_index = 1  # Frame_index 1 has access to the true ego speed
+        ego_input = self.sim_dataset.rasterise_frame_batch(self.frame_index)
+        self.ego_input_dict = convert_to_numpy(ego_input[0])
+
+        # Reset Kinematic model
+        if self.use_kinematic:
+            init_kin_state = np.array([0.0, 0.0, 0.0, 0.1 * ego_input[0]['curr_speed']])
```

Comment:
why multiply by 0.1?

Reply:
Because the step_time is 0.1. I have added a comment there.